### PR TITLE
Update status of course

### PR DESCRIPTION
**Pull request by Tranchinhthuc**
Update status of course

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -6,4 +8,11 @@ class Course < ActiveRecord::Base
 
   validates :title, presence: true
   validates :description, presence: true
+
+  enum status: ["Ready", "Active", "Finished"]
```
Comment (ThuBM) [inline app/models/course.rb]: "Ready"  =>  :ready
"Active"  =>  :active
"Finished" => :finished

File: app/models/course.rb [hunk 2]
```diff
@@ -6,4 +8,11 @@ class Course < ActiveRecord::Base
...
+
+  private
+  def set_default_status
+    self.role = I18n.t("application.course_status.ready") unless self.status.present?
```
Comment (ThuBM) [inline app/models/course.rb]: self.role = :ready  if  ...

**Comment by ThuBM** [comment]
ok

File: app/models/ability.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ def initialize user
         (current_user == user) || (user.classmate_with? current_user)
       end
       can :show, Course do |course|
-        course.try(:users).include? user
+        (course.try(:users).include? user) && (course.active?)
```
Comment (dieunb) [inline app/models/ability.rb]: course.users.include?(user) && course.active?

**Pull request merged**